apps/cowswap-frontend/src/modules/limitOrders/containers/LimitOrdersWarnings/index.tsx

```diff
@@ -99,14 +107,25 @@ export function LimitOrdersWarnings(props: LimitOrdersWarningsProps) {
   const { state } = useDerivedTradeState()
   const showRecipientWarning = isConfirmScreen && state?.recipient && account !== state.recipient
 
+  const native = useNativeCurrency()
+  const wrapped = useWrappedToken()
+  const isNativeIn = useIsNativeIn()
+  const isWrappedOut = useIsWrappedOut()
+
+  // TODO: implement Safe App EthFlow bundling for LIMIT and disable the warning in that case
+  const showNativeSellWarning = isNativeIn && !isWrappedOut && native && wrapped
+
+  const navigateOnCurrencySelection = useNavigateOnCurrencySelection()
```

Comment:
Since the code is duplicated I would suggest creating of smart component wrapper for `SellNativeWarningBanner` that will collect all data for the pure component